src/core/server/saved_objects/service/lib/repository_es_client.ts

```diff
@@ -14,11 +14,13 @@ import { decorateEsError } from './decorate_es_error';
 
 const methods = [
   'bulk',
+  'closePointInTime',
```

Comment:
Is this currently used anywhere?

Reply:
Not yet, need to figure out a plan for how we want to deal with closing a PIT. I don't love any of the options right now:

1. Do we simply let the keep_alive expire?
2. Do we expect the consumer to call `closePointInTime` manually via ES client?
3. Do we add another method to SO client to close PIT, which would literally do nothing but call ES client under the hood?

(1) Feels like a risky thing to do (what if someone opens a PIT with a long keep_alive and fails to close it? Then we are consuming unnecessary resources)
(2) Is a pain for the consumer because now they need ES client after all (one of the reasons we went with `openPointInTimeForType` on the SO client was so that consumers wouldn't be in this situation)
(3) Feels a bit unnecessary since this method would have no advantages - it just makes the call to the ES client implicit.

WDYT?

Reply:
> Do we simply let the keep_alive expire?

Reply:
> I don't love any of the options right now:

Agreed, unfortunately...

> 1. Do we simply let the keep_alive expire?

Even considering how easy doing that would be, I guess we can't consider this as a valid option

> 2. Do we expect the consumer to call closePointInTime manually via ES client?

I think it wouldn't make such sense given that we decided to go with a specific API for `openPointInTimeForType`

> 3. Do we add another method to SO client to close PIT, which would literally do nothing but call ES client under the hood?

That's the most consistent approach imho considering we decided to go with the `openPointInTimeForType` API.

Now, thinking out loud:

- Would auto-closing the PIT once we reach the end of the results (do we have such information when using the `pit` option with `search` btw) be an acceptable solution? I guess we could imagine some situations where the consumer want to 'reuse' the PIT to re-fetch results after some specific `searchAfter`, so it seems fragile

- Could we introduce a `closePit` option to `find` so consumers can decide when to close the PIT. I don't think it would be useful, as consumers needs to read the results to know if they want to close the PIT, so I guess this is worthless. 

- (Mix of both) Introduce a `closePitIfComplete` instead, that would close the PIT if we reached the end of the results

Note that options are closer to the `openPit` option approach than the `openPointInTimeForType` we decided to go with. If we add a `closePit(IfComplete)` options, we might consider changing our implementation to the `openPit` one

Overall, I still feel like the `Do we add another method to SO client to close PIT, which would literally do nothing but call ES client under the hood` may be the most straightforward and easiest implementation, and would be good enough for now. We could eventually reconsider than once we see actual usages of this new API being implemented?

cc @rudolf

Reply:
Even if for the saved objects indices the impact might be fairly low because the index is usually small it's hard to estimate the impact when e.g. task manager/alerting is heavily writing while there's many open PIT's. I think we should assume a PIT is a resource hog which needs to be used carefully and cleaned up.

For scroll searches it was fairly easy to open too many scroll contexts and then get failures because of "too many open scroll contexts" I'd be surprised if there isn't a similar "circuit breaker" for open PIT's. So cleaning up the PIT immediately after a consumer is done would be a good idea.

I favour (3) just for the developer experience.

Reply:
I also think if there's an obvious way to improve the developer experience we should push for the Elasticsearch API to be improved. E.g. if we think an auto-closing PIT which automatically closes once all results have been paged through would solve a lot of DX problems, we should first push for this to be added to ES.

Not to say we can't add _any_ sugar to the API, but anything we do in saved objects is probably going to be harder to get right and guarantee it doesn't break in subtle ways and harder to maintain than an upstream fix.